modules/routing/src/router.ts

```diff
@@ -14,26 +15,93 @@ type Route = {
 	httpMethod: HttpMethod;
 	path: string;
 	handler: (event: APIGatewayProxyEvent) => Promise<APIGatewayProxyResult>;
+	validation?: {
+		path?: z.Schema;
+		body?: z.Schema;
+	};
 };
 
 export const NotFoundResponse = {
 	body: 'Not Found',
 	statusCode: 404,
 };
 
+function matchPath(routePath: string, eventPath: string): { matched: boolean; params: Record<string, string> } {
```

Comment:
would it make more sense to return a sum type (ie make it optional).
Then we can make it clear from the types that matched=false and params have a value is not possible
```suggestion
function matchPath(routePath: string, eventPath: string): params: Record<string, string> | undefined {
```

Reply:
New pull request created just for the changes on router module -> https://github.com/guardian/support-service-lambdas/pull/2886